.github/actions/test-coverage/determine_test_coverage.py

```diff
@@ -0,0 +1,132 @@
+import ast
+import importlib.util
+import sys
+from pathlib import Path
+import json
+import math
+
+
+class TestCoverageError(Exception):
+    pass
```

Comment:
Let's make this a little friendlier with a few edits:

```suggestion
class TestCoverageError(Exception):
    
    def __init__(self, percent: int, actions_missing_coverage: Iterable[str]) -> None:
        self.percent = percent
        self.actions_missing_coverage = sorted(actions_missing_coverage)
        
    def __str__(self) -> str:
         action_str = "\n".join(f"    - {action}" for action in self.actions_missing_coverage)
        return f"Only {self.percent}% of actions have tests. These actions appear to be missing tests: {action_str}"
```